### PR TITLE
Check that a node is in fact a travelnet or elevator before parsing input data

### PR DESCRIPTION
**Pull request by oversword**
Created a new `travelnet.is_travelnet_or_elevator` method for checking if the node at a position is in fact dealt with by this mod and used this to validate the `pos2str` value.

This may be made redundant by the resolution to other issues (#33), but ATM it's a crashing bug, and the new API method is probably nice to have anyways.

Logic copied from the destination check.

### CODE REVIEW

File: functions.lua [hunk 1]
```diff
@@ -80,6 +80,12 @@ function travelnet.is_elevator(node_name)
 	return node_name == "travelnet:elevator"
 end
 
+function travelnet.is_travelnet_or_elevator(pos)
+	local node = minetest.get_node(pos)
+	local node_def = minetest.registered_nodes[node.name]
+	return node_def.groups.travelnet or node_def.groups.elevator
```
Comment (S-S-X) [inline functions.lua:86]: is it guaranteed that every possible node has groups table, also ignore nodes?

if not or not sure then maybe
```suggestion
	return node_def and node_def.groups and (node_def.groups.travelnet or node_def.groups.elevator)
```
node_def should be checked anyway (for unknown nodes) so at least
```suggestion
	return node_def and (node_def.groups.travelnet or node_def.groups.elevator)
```

Comment (oversword) [reply functions.lua:86]: Oh yeah I just checked, if you register it without groups it won't have any, best to check both

**Review by S-S-X — APPROVED** [review]
LGTM, also fixes few old and rare but possible corner case crashes.

**Comment by wsor4035** [comment]
seems like a over complication of `minetest.get_item_group(itemname, "travelnet") > 0 or minetest.get_item_group(itemname, "elevator") > 0`

**Comment by S-S-X** [comment]
> seems like a over complication of `minetest.get_item_group(itemname, "travelnet") > 0 or minetest.get_item_group(itemname, "elevator") > 0`

Personally I don't like that but yes it seems to be better way.

Also checking node_def.groups seems unnecessary, empty table is provided through metatable __index.

**Pull request merged**